src/app/admin/components/sidebar/sidebar.component.ts

```diff
@@ -1,25 +1,39 @@
-import { Component, OnInit } from '@angular/core';
-import { Location } from '@angular/common';
+import { Component, OnInit, OnDestroy, Input } from '@angular/core';
+import { Router } from '@angular/router';
+import { Subject } from 'rxjs';
+import { increaseElementDepthCount } from '@angular/core/src/render3/state';
 
 @Component({
   selector: 'clark-admin-sidebar',
   templateUrl: './sidebar.component.html',
   styleUrls: ['./sidebar.component.scss']
 })
-export class SidebarComponent implements OnInit {
+export class SidebarComponent implements OnInit, OnDestroy {
+  destroyed$: Subject<void> = new Subject();
 
-  constructor(private location: Location) { }
+  @Input() authorizedCollections: string[] = [];
```

Comment:
See above comment, but I think this would be better named as `collections` since the component is not dealing with anything more than the collections it is given.